### PR TITLE
Feature + Fix: Hoppity/CF Updates

### PR DESCRIPTION
**Pull request by DavidArthurCole**
## Dependencies
- https://github.com/hannibal002/SkyHanni/pull/2925
- https://github.com/hannibal002/SkyHanni-REPO/pull/293

## What
Adds a bunch of new features relating to the new stuff from Hoppity/CF update.
This also migrated all Hoppity Collection stat config options to their own config, and handled those migrations. While I was there, I reorganized the strewn Accordions within Inventory > Chocolate Factory.

<details>
<summary>Images</summary>

</details>

## Changelog New Features
+ Added a 30-second stay timer to Chocolate Factory after a Meal Egg is opened. - Daveed
+ Added Hoppity Event stat text for Hitman Rabbits. - Daveed
+ Added Hoppity Collection highlighting for Resident and Hotspot Rabbits. - Daveed
    * Highlights Resident/Hotspot Rabbits on your current island.
+ Added Resident and Hotspot Rabbit overview option to Hoppity Collection stats. - Daveed
+ Added Hitman slot tracking to display the most recent rabbits you obtained. - Daveed
    * Displays slots with an active cooldown.
+ Added an option to show the Hoppity Event Card only while on islands where eggs spawn. - Daveed

## Changelog Fixes
+ Fixed Hoppity Meals being marked as spawned immediately when the Hoppity Hunt started. - Daveed
+ Fixed an issue with Hoppity Chat Compact duplicating El Dorado messages. - Daveed
+ Fixed "Fish the Rabbit" being detected only on duplicate instances. - Daveed

### CODE REVIEW

**Comment by DavidArthurCole** [comment]
Update tomorrow, this is ready (assuming detekt passes 😔)

**Comment by hannibal002** [comment]
leftToPurchase: 19

image

File: src/main/java/at/hannibal2/skyhanni/features/event/hoppity/HoppityEventSummary.kt [hunk 1]
```diff
@@ -82,6 +85,22 @@ object HoppityEventSummary {
     private val storage get() = ProfileStorageData.profileSpecific
     private val liveDisplayConfig get() = config.eventSummary.liveDisplay
     private val updateCfConfig get() = config.eventSummary.cfReminder
+    private val disallowListHoppityIslands by lazy {
```
Comment (hannibal002) [inline src/main/java/at/hannibal2/skyhanni/features/event/hoppity/HoppityEventSummary.kt]: cant we use a list of allowd islands instead, and load that from the repo via the egg location data? or is this check different here?

**Review by hannibal002 — CHANGES_REQUESTED** [review]
cant test the active hoppity stuff but looks fine overall

**Pull request merged**